### PR TITLE
#1311 check existing email before user creation

### LINKED ISSUE #1311: Check for existing email before user creation
When deleting a user (currently) it sets them to inactive (active=0) – however if you go to create a new user with the same unique email address it lets you but then throws a "unique" error.

`(directus_users_email_unique)`

We should check if an email already exists when trying to create a new user – and offer to TURN ON that user instead.

### PR DESCRIPTION
**Pull request by Toothless1118**
I fixed #1311 issue, please check

### CODE REVIEW

File: api/api.php [hunk 1]
```diff
@@ -933,8 +933,20 @@
 
     switch ($app->request()->getMethod()) {
         case 'POST':
-            $newRecord = $entriesService->createEntry($table, $payload, $params);
-            $params[$tableGateway->primaryKeyFieldName] = $newRecord[$tableGateway->primaryKeyFieldName];
+            if ($table == 'directus_users') {
```
Comment (WellingGuzman) [inline api/api.php]: Also this looks good.

Can you check https://github.com/directus/directus/blob/master/api/routes/A1/Entries.php#L23 and implement it as well for **API 1.1 for 6.4 on version/6.4 branch**

**Review by wellingguzman — CHANGES_REQUESTED** [review]
This is looking good, would be awesome if time exists, that you could add it to API 1.1 and Directus 6.4, otherwise let me know so I could merge it

File: .gitignore [hunk 2]
```diff
@@ -34,4 +34,7 @@ customs/uis/*
 .gulp-scss-cache
 node_modules
 storage
+bin/composer
```
Comment (Toothless1118) [inline .gitignore]: These files created when installing composer.
php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');"
php -r "if (hash_file('SHA384', 'composer-setup.php') === '669656bab3166a7aff8a7506b8cb2d1c292f042046c5a994c43155c0be6190fa0355160742ab2e1c88d40d5be660b410') { echo 'Installer verified'; } else { echo 'Installer corrupt'; unlink('composer-setup.php'); } echo PHP_EOL;"

Comment (WellingGuzman) [inline .gitignore]: Thanks, it won't hurt, but we shouldn't be tracking (in this case untracking) those, the person installing should clean the directory after installation, and I believe the composer-setup clean all this.

File: api/api.php (hunk 1 above)
Comment (Toothless1118) [inline api/api.php]: I have implemented it on Entries.php and pushed it again. Please check.

**Comment by WellingGuzman** [comment]
Hey @Toothless1118 thanks for the contribution.

I'm wondering if you are going to complete the whole issue from https://github.com/directus/directus/issues/1311 or are just going to complete the server-side creating the update or create approach?

**Pull request closed without merging**